<details><summary>before (on master)</summary><p>

```
$ test/functional/feature_nulldummy.py --valgrind -l debug
2021-03-05T15:19:52.679000Z TestFramework (DEBUG): PRNG seed is: 7345676661789155999
2021-03-05T15:19:52.679000Z TestFramework (DEBUG): Setting up network thread
2021-03-05T15:19:52.680000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_0_okgkoj
2021-03-05T15:19:52.684000Z TestFramework.node0 (DEBUG): bitcoind started, waiting for RPC to come up
2021-03-05T15:19:52.688000Z TestFramework.node1 (DEBUG): bitcoind started, waiting for RPC to come up
2021-03-05T15:20:12.471000Z TestFramework.node0 (DEBUG): RPC successfully started
2021-03-05T15:20:12.823000Z TestFramework.node1 (DEBUG): RPC successfully started
2021-03-05T15:20:20.161000Z TestFramework.node0 (DEBUG): TestNode.generate() dispatches `generate` call to `generatetoaddress`
2021-03-05T15:20:21.485000Z TestFramework.node0 (DEBUG): TestNode.generate() dispatches `generate` call to `generatetoaddress`
2021-03-05T15:20:51.514000Z TestFramework (ERROR): JSONRPC error
Traceback (most recent call last):
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/authproxy.py", line 168, in _get_response
    http_response = self.__conn.getresponse()
  File "/usr/lib/python3.9/http/client.py", line 1347, in getresponse
    response.begin()
  File "/usr/lib/python3.9/http/client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.9/http/client.py", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/lib/python3.9/socket.py", line 704, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 128, in main
    self.run_test()
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/feature_nulldummy.py", line 71, in run_test
    self.nodes[0].generate(427)  # Block 429
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_node.py", line 302, in generate
    return self.generatetoaddress(nblocks=nblocks, address=self.get_deterministic_priv_key().address, maxtries=maxtries)
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/authproxy.py", line 144, in __call__
    response, status = self._request('POST', self.__url.path, postdata.encode('utf-8'))
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/authproxy.py", line 108, in _request
    return self._get_response()
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/authproxy.py", line 170, in _get_response
    raise JSONRPCException({
test_framework.authproxy.JSONRPCException: 'generatetoaddress' RPC took longer than 30.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)
2021-03-05T15:20:51.519000Z TestFramework (DEBUG): Closing down network thread
2021-03-05T15:20:51.570000Z TestFramework (INFO): Stopping nodes
2021-03-05T15:20:51.570000Z TestFramework.node0 (DEBUG): Stopping node
2021-03-05T15:20:51.571000Z TestFramework.node0 (ERROR): Unable to stop node.
Traceback (most recent call last):
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_node.py", line 323, in stop_node
    self.stop(wait=wait)
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/authproxy.py", line 144, in __call__
    response, status = self._request('POST', self.__url.path, postdata.encode('utf-8'))
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/authproxy.py", line 107, in _request
    self.__conn.request(method, path, postdata, headers)
  File "/usr/lib/python3.9/http/client.py", line 1255, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.9/http/client.py", line 1266, in _send_request
    self.putrequest(method, url, **skips)
  File "/usr/lib/python3.9/http/client.py", line 1092, in putrequest
    raise CannotSendRequest(self.__state)
http.client.CannotSendRequest: Request-sent
2021-03-05T15:20:51.572000Z TestFramework.node1 (DEBUG): Stopping node
2021-03-05T15:21:51.829000Z TestFramework.utils (ERROR): wait_until() failed. Predicate: ''''
    def is_node_stopped(self):
        """Checks whether the node has stopped.

        Returns True if the node has stopped. False otherwise.
        This method is responsible for freeing resources (self.process)."""
        if not self.running:
            return True
        return_code = self.process.poll()
        if return_code is None:
            return False

        # process has stopped. Assert that it didn't return an error code.
        assert return_code == 0, self._node_msg(
            "Node returned non-zero exit code (%d) when stopping" % return_code)
        self.running = False
        self.process = None
        self.rpc_connected = False
        self.rpc = None
        self.log.debug("Node stopped")
        return True
'''
Traceback (most recent call last):
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/feature_nulldummy.py", line 134, in <module>
    NULLDUMMYTest().main()
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 151, in main
    exit_code = self.shutdown()
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 294, in shutdown
    self.stop_nodes()
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 545, in stop_nodes
    node.wait_until_stopped()
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_node.py", line 369, in wait_until_stopped
    wait_until_helper(self.is_node_stopped, timeout=timeout, timeout_factor=self.timeout_factor)
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/util.py", line 261, in wait_until_helper
    raise AssertionError("Predicate {} not true after {} seconds".format(predicate_source, timeout))
AssertionError: Predicate ''''
    def is_node_stopped(self):
        """Checks whether the node has stopped.

        Returns True if the node has stopped. False otherwise.
        This method is responsible for freeing resources (self.process)."""
        if not self.running:
            return True
        return_code = self.process.poll()
        if return_code is None:
            return False

        # process has stopped. Assert that it didn't return an error code.
        assert return_code == 0, self._node_msg(
            "Node returned non-zero exit code (%d) when stopping" % return_code)
        self.running = False
        self.process = None
        self.rpc_connected = False
        self.rpc = None
        self.log.debug("Node stopped")
        return True
''' not true after 60.0 seconds
[node 1] Cleaning up leftover process
[node 0] Cleaning up leftover process
```
</p></details>

<details><summary>after (with this patch)</summary><p>

```
$ test/functional/feature_nulldummy.py --valgrind -l debug
2021-03-05T15:40:11.700000Z TestFramework (DEBUG): PRNG seed is: 1307129330788481038
2021-03-05T15:40:11.700000Z TestFramework (DEBUG): Setting up network thread
2021-03-05T15:40:11.700000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_qdtmppe1
2021-03-05T15:40:11.704000Z TestFramework.node0 (DEBUG): bitcoind started, waiting for RPC to come up
2021-03-05T15:40:11.708000Z TestFramework.node1 (DEBUG): bitcoind started, waiting for RPC to come up
2021-03-05T15:40:32.086000Z TestFramework.node0 (DEBUG): RPC successfully started
2021-03-05T15:40:32.454000Z TestFramework.node1 (DEBUG): RPC successfully started
2021-03-05T15:40:40.177000Z TestFramework.node0 (DEBUG): TestNode.generate() dispatches `generate` call to `generatetoaddress`
2021-03-05T15:40:41.407000Z TestFramework.node0 (DEBUG): TestNode.generate() dispatches `generate` call to `generatetoaddress`
2021-03-05T15:42:09.617000Z TestFramework (INFO): Test 1: NULLDUMMY compliant base transactions should be accepted to mempool and mined before activation [430]
2021-03-05T15:42:13.159000Z TestFramework (INFO): Test 2: Non-NULLDUMMY base multisig transaction should not be accepted to mempool before activation
2021-03-05T15:42:13.563000Z TestFramework (INFO): Test 3: Non-NULLDUMMY base transactions should be accepted in a block before activation [431]
2021-03-05T15:42:14.022000Z TestFramework (INFO): Test 4: Non-NULLDUMMY base multisig transaction is invalid after activation
2021-03-05T15:42:14.593000Z TestFramework (INFO): Test 5: Non-NULLDUMMY P2WSH multisig transaction invalid after activation
2021-03-05T15:42:15.205000Z TestFramework (INFO): Test 6: NULLDUMMY compliant base/witness transactions should be accepted to mempool and in block after activation [432]
2021-03-05T15:42:15.798000Z TestFramework (DEBUG): Closing down network thread
2021-03-05T15:42:15.849000Z TestFramework (INFO): Stopping nodes
2021-03-05T15:42:15.849000Z TestFramework.node0 (DEBUG): Stopping node
2021-03-05T15:42:15.895000Z TestFramework.node1 (DEBUG): Stopping node
2021-03-05T15:42:18.436000Z TestFramework.node0 (DEBUG): Node stopped
2021-03-05T15:42:19.288000Z TestFramework.node1 (DEBUG): Node stopped
2021-03-05T15:42:19.288000Z TestFramework (INFO): Cleaning up /tmp/bitcoin_func_test_qdtmppe1 on exit
2021-03-05T15:42:19.288000Z TestFramework (INFO): Tests successful
```
</p></details>

--- a/test/functional/feature_nulldummy.py
+++ b/test/functional/feature_nulldummy.py
@@ -6,11 +6,11 @@
 
 Connect to a single node.
 Generate 2 blocks (save the coinbases for later).
-Generate 427 more blocks.
-[Policy/Consensus] Check that NULLDUMMY compliant transactions are accepted in the 430th block.
+Generate COINBASE_MATURITY (CB) more blocks to ensure the coinbases are mature.
+[Policy/Consensus] Check that NULLDUMMY compliant transactions are accepted in block CB + 3.
 [Policy] Check that non-NULLDUMMY transactions are rejected before activation.
-[Consensus] Check that the new NULLDUMMY rules are not enforced on the 431st block.
-[Policy/Consensus] Check that the new NULLDUMMY rules are enforced on the 432nd block.
+[Consensus] Check that the new NULLDUMMY rules are not enforced on block CB + 4.
+[Policy/Consensus] Check that the new NULLDUMMY rules are enforced on block CB + 5.
 """
 import time
 
@@ -20,6 +20,7 @@ from test_framework.script import CScript
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import assert_equal, assert_raises_rpc_error
 
+COINBASE_MATURITY = 427
 NULLDUMMY_ERROR = "non-mandatory-script-verify-flag (Dummy CHECKMULTISIG argument must be zero)"
 
 def trueDummy(tx):
@@ -43,7 +44,7 @@ class NULLDUMMYTest(BitcoinTestFramework):
         # This script tests NULLDUMMY activation, which is part of the 'segwit' deployment, so we go through
         # normal segwit activation here (and don't use the default always-on behaviour).
         self.extra_args = [[
-            '-segwitheight=432',
+            f'-segwitheight={COINBASE_MATURITY + 5}',
             '-addresstype=legacy',
         ]] * 2
 
@@ -64,16 +65,16 @@ class NULLDUMMYTest(BitcoinTestFramework):
             wmulti.importaddress(self.ms_address)
             wmulti.importaddress(self.wit_ms_address)
 
-        self.coinbase_blocks = self.nodes[0].generate(2)  # Block 2
+        self.coinbase_blocks = self.nodes[0].generate(2)  # block height = 2
         coinbase_txid = []
         for i in self.coinbase_blocks:
             coinbase_txid.append(self.nodes[0].getblock(i)['tx'][0])
-        self.nodes[0].generate(427)  # Block 429
+        self.nodes[0].generate(COINBASE_MATURITY)  # block height = COINBASE_MATURITY + 2
         self.lastblockhash = self.nodes[0].getbestblockhash()
-        self.lastblockheight = 429
-        self.lastblocktime = int(time.time()) + 429
+        self.lastblockheight = COINBASE_MATURITY + 2
+        self.lastblocktime = int(time.time()) + self.lastblockheight
 
-        self.log.info("Test 1: NULLDUMMY compliant base transactions should be accepted to mempool and mined before activation [430]")
+        self.log.info(f"Test 1: NULLDUMMY compliant base transactions should be accepted to mempool and mined before activation [{COINBASE_MATURITY + 3}]")
         test1txs = [create_transaction(self.nodes[0], coinbase_txid[0], self.ms_address, amount=49)]
         txid1 = self.nodes[0].sendrawtransaction(test1txs[0].serialize_with_witness().hex(), 0)
         test1txs.append(create_transaction(self.nodes[0], txid1, self.ms_address, amount=48))
@@ -87,7 +88,7 @@ class NULLDUMMYTest(BitcoinTestFramework):
         trueDummy(test2tx)
         assert_raises_rpc_error(-26, NULLDUMMY_ERROR, self.nodes[0].sendrawtransaction, test2tx.serialize_with_witness().hex(), 0)
 
-        self.log.info("Test 3: Non-NULLDUMMY base transactions should be accepted in a block before activation [431]")
+        self.log.info(f"Test 3: Non-NULLDUMMY base transactions should be accepted in a block before activation [{COINBASE_MATURITY + 4}]")
         self.block_submit(self.nodes[0], [test2tx], False, True)
 
         self.log.info("Test 4: Non-NULLDUMMY base multisig transaction is invalid after activation")
@@ -104,7 +105,7 @@ class NULLDUMMYTest(BitcoinTestFramework):
         assert_raises_rpc_error(-26, NULLDUMMY_ERROR, self.nodes[0].sendrawtransaction, test5tx.serialize_with_witness().hex(), 0)
         self.block_submit(self.nodes[0], [test5tx], True)
 
-        self.log.info("Test 6: NULLDUMMY compliant base/witness transactions should be accepted to mempool and in block after activation [432]")
+        self.log.info(f"Test 6: NULLDUMMY compliant base/witness transactions should be accepted to mempool and in block after activation [{COINBASE_MATURITY + 5}]")
         for i in test6txs:
             self.nodes[0].sendrawtransaction(i.serialize_with_witness().hex(), 0)
         self.block_submit(self.nodes[0], test6txs, True, True)
@@ -129,6 +130,7 @@ class NULLDUMMYTest(BitcoinTestFramework):
             self.lastblockheight += 1
         else:
             assert_equal(node.getbestblockhash(), self.lastblockhash)
+
 
 if __name__ == '__main__':
     NULLDUMMYTest().main()

--- a/test/functional/feature_nulldummy.py
+++ b/test/functional/feature_nulldummy.py
@@ -20,14 +20,14 @@ from test_framework.script import CScript
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import assert_equal, assert_raises_rpc_error
 
-COINBASE_MATURITY = 427
+COINBASE_MATURITY = 100
 NULLDUMMY_ERROR = "non-mandatory-script-verify-flag (Dummy CHECKMULTISIG argument must be zero)"
 
 def trueDummy(tx):
     scriptSig = CScript(tx.vin[0].scriptSig)
     newscript = []
     for i in scriptSig:
-        if (len(newscript) == 0):
+        if len(newscript) == 0:
             assert len(i) == 0
             newscript.append(b'\x51')
         else:
@@ -38,7 +38,7 @@ def trueDummy(tx):
 class NULLDUMMYTest(BitcoinTestFramework):
 
     def set_test_params(self):
-        # Need two nodes only so GBT doesn't complain that it's not connected
+        # Need two nodes so GBT (getblocktemplate) doesn't complain that it's not connected.
         self.num_nodes = 2
         self.setup_clean_chain = True
         # This script tests NULLDUMMY activation, which is part of the 'segwit' deployment, so we go through

